Clarify NetworkStack porting guide

SeppoTakalo opened the pull request:
Socket porting part was missing information about non-blocking
behaviour of the underlying stack calls.

Connect call was not mentioning the default interfaces at all.

Testing section was having outdated information. Replaced with the
link to current test plan.

Modify the ESP case study with up to date JSON and test result.

SeppoTakalo commented:
@kjbracey-arm Please review.

SeppoTakalo commented:
@AnotherButler Please review.

AnotherButler commented:
ping @kjbracey-arm

kjbracey-arm commented on docs/porting/connectivity/NetworkStack.md:
```diff
@@ -25,106 +32,56 @@ Each subclass has distinct pure virtual methods. Visit their class references (l
+High-level API calls to an implementation of a network-socket API are **identical** across networking protocols. The only difference is the interface object constructor and the method through which you connect to the network. For example, a Wi-Fi connection requires an SSID and password, a cellular connection requires an APN and Ethernet doesn't require any credentials. Each interface type may pr ...
 
-Below is a demonstration with the code that sends an HTTP request over Ethernet:
+Below is a demonstration with the code that connects to a network and relies that required configuration are provided:
```
"relies on"

kjbracey-arm commented on docs/porting/connectivity/NetworkStack.md:
```diff
@@ -25,106 +32,56 @@ Each subclass has distinct pure virtual methods. Visit their class references (l
+#### Errors
 
-Many functions of `NetworkStack` and `NetworkInterface` have return types of `nsapi_error_t`, which is a type used to represent error codes. You can see a list of these return codes [here](https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.4/api/group__netsocket.html#gac21eb8156cf9af198349069cdc7afeba). You can view the integer values the error macros in [this file](https://github.com/ARMmbed/mb ...
+Many functions of `NetworkStack` and `NetworkInterface` have return types of `nsapi_error_t`, which is a type used to represent error codes. You can see a list of these return codes [here](https://os.mbed.com/docs/v5.10/mbed-os-api-doxy/group__netsocket.html#gac21eb8156cf9af198349069cdc7afeba). You can view the integer values the error macros in [this file](https://github.com/ARMmbed/mbed-os/blob ...
```
Some calls like `recv` can also return non-negative size values as success - these use `nsapi_size_or_error_t` as their return type.

SeppoTakalo commented:
@AnotherButler changes seem OK.

Added one clarification regarding Kevin's comment on positive return values.

Also rebased this on top of `development`

SeppoTakalo commented:
CI job is broken
```
The command "E: Failed to fetch https://dl.hhvm.com/ubuntu/dists/trusty/main/binary-amd64/Packages server certificate verification failed. CAfile: /etc/ssl/certs/ca-certificates.crt CRLfile: none" failed. Retrying, 2 of 3.
```

Who maintains this?

AnotherButler commented:
This is actually the same Travis script that `mbed-os` uses. It appears to have been a spurious failure.
